test/Dialect/FIRRTL/parse-basic.fir

```diff
@@ -1111,6 +1111,17 @@ circuit MyModule :     ; CHECK: firrtl.circuit "MyModule" {
     ; CHECK-NEXT: [[NOT_COND:%.+]] = firrtl.not %cond
     ; CHECK-NEXT: firrtl.assert %clock, [[NOT_COND]], [[NOT_ENABLE]]
 
+    ; Verification Library Covers
+
+    ; Predicate modifier `noMod`
+    when cond:
+      printf(clock, enable, "Assertion failed: [verif-library-cover]<extraction-summary>{\"predicateModifier\":{\"type\":\"noMod\"},\"conditionalCompileToggles\":[{\"type\":\"unrOnly\"},{\"type\":\"formalOnly\"}],\"labelExts\":[\"cover\",\"label\"],\"format\":{\"type\":\"sva\"},\"baseMsg\":\"cover hello world\"}", value)
+      stop(clock, enable, 1)
+    ; CHECK: [[TMP:%.+]] = firrtl.not %cond
```

Comment:
I think this is wrong. Is the cover predicate supposed to be inverted?

Reply:
I'm not really sure where the first inversion comes from, but it ends up with the correct polarity when I go down to verilog. The same inversion exists in assert/assume — not sure if you know where those come from?

Reply:
#3219 has some context here, touches the places the inversion happens and in particular handles Cover separately from the others.  Checking output verilog is good, particularly the way enable+predicate are combined.

Reply:
This Verilog looks wrong to me, or maybe it works coming from Chisel?

The following is a FIRRTL circuit of the above where the condition is un-inverted:

```
circuit Foo:
  module Foo:
    input clock: Clock
    input cond: UInt<1>
    input enable: UInt<1>
    input value: UInt<8>

    when cond:
      printf(clock, enable, "Assertion failed: [verif-library-cover]<extraction-summary>{\"predicateModifier\":{\"type\":\"noMod\"},\"conditionalCompileToggles\":[{\"type\":\"unrOnly\"},{\"type\":\"formalOnly\"}],\"labelExts\":[\"cover\",\"label\"],\"format\":{\"type\":\"sva\"},\"baseMsg\":\"cover hello world\"}", value)
      stop(clock, enable, 1)
```

In the output Verilog, this is covering `enable & ~cond`:

```verilog
module Foo(
  input       clock,
              cond,
              enable,
  input [7:0] value);

  `ifdef USE_UNR_ONLY_CONSTRAINTS
    `ifdef USE_FORMAL_ONLY_CONSTRAINTS
      cover__verif_library_cover_label: cover property (@(posedge clock) enable & ~cond);
    `endif // USE_FORMAL_ONLY_CONSTRAINTS
  `endif // USE_UNR_ONLY_CONSTRAINTS
endmodule
```

I think this should be `enable & cond`, yes?

Reply:
I feel like what I am seeing going from Chisel to verilog is not lining up with what you are seeing 😕 

Chisel
```scala
    // cover with predicate only
    cover(sum === 0.U)
```

FIRRTL
```
    node _T = eq(sum, UInt<1>("h0")) @[ExtractCoversSpec.scala 51:15]
    node _T_1 = asUInt(reset) @[ExtractCoversSpec.scala 51:10]
    node _T_2 = eq(_T_1, UInt<1>("h0")) @[ExtractCoversSpec.scala 51:10]
    when _T_2 : @[ExtractCoversSpec.scala 51:10]
      node _T_3 = eq(_T, UInt<1>("h0")) @[ExtractCoversSpec.scala 51:10]
      when _T_3 : @[ExtractCoversSpec.scala 51:10]
        printf(clock, UInt<1>("h1"), "Assertion failed: [verif-library-cover]<extraction-summary>{\"predicateModifier\":{\"type\":\"noMod\"},\"format\":{\"type\":\"ifElseFatal\"},\"baseMsg\":\"cover failed (verification library):   @[ExtractCoversSpec.scala 51:10]\"}</extraction-summary>\n    at Property.scala:96 chisel3.assert(predicate, flag + exStr, msgArgs: _*)(sourceInfo, compileOptions)\n") : printf @[ExtractCoversSpec.scala 51:10]
      assert(clock, _T, UInt<1>("h1"), "") : assert @[ExtractCoversSpec.scala 51:10]
```

Verilog
```verilog
cover__verif_library: cover property (@(posedge clock) ~reset & sum == 8'h0);	// ExtractCoversSpec.scala:46:37, :51:{10,15}
```

Reply:
i think there is another inversion that is coming in between Chisel and FIRRTL:
``` 
node _T = eq(sum, UInt<1>("h0")) @[ExtractCoversSpec.scala 51:15]
```
and
```
node _T_3 = eq(_T, UInt<1>("h0")) @[ExtractCoversSpec.scala 51:10]
```

Reply:
Alright, so the internal verification library will emit this as:

```
when not(cond):
  printf(..., "Assertion failed: [veriif-library-cover]")
assert(..., "")
```

If this is what is going to happen, can the test be rewritten to show this?

Also, is the `assert` getting removed by MFC?

Reply:
`CHECK-NOT` will guarantee that something is not there.